src/main/java/com/jfrog/xray/client/impl/XrayClient.java

```diff
@@ -108,9 +112,20 @@ private String createUrl(String queryPath) {
     }
 
     private CloseableHttpResponse setHeadersAndExecute(HttpUriRequest request) throws IOException {
-        CloseableHttpResponse response = execute(request);
+        HttpClientContext context = HttpClientContext.create();
+        CloseableHttpResponse response = execute(request, context);
         StatusLine statusLine = response.getStatusLine();
         int statusCode = statusLine.getStatusCode();
+
+        List<URI> locations = context.getRedirectLocations();
+        // Redirection indication
+        if (locations != null) {
+            for (URI uri : locations) {
+                if (uri.toString().contains("reactivate-server")) {
+                    throw new JFrogInactiveEnvironmentException(statusCode, "JFrog Platform is inactive", uri.toString());
+                }
+            }
+        }
```

Comment:
I think that this logic deserves exporting to a function "handleRedirections" or something similar.